### DISCUSSION TITLE
What is default expiry, If we don't specify expiry while writing data to cache?

### DISCUSSION THREAD

**Discussion by rahulradixweb** [TruCopilot/phpfastcache#834] [Q&A] [answered]
Hi,

We have implemented file driver for caching data, something like below.

```
Phpfastcache\CacheManager::setDefaultConfig(new Phpfastcache\Config\ConfigurationOption([
    "path" => "/tmp/",
]));
$InstanceCache = Phpfastcache\CacheManager::getInstance('files');
$cachedItem = $InstanceCache->getItem('sample_cache_key');
$cachedData['cache_key'] = "cache_value";
$cachedItem->set($cachedData)->addTag('sample_tag'); // We don't specify expiry here
$InstanceCache->save($cachedItem);
```
**My question is , up-to when this data will be available in cached storage?**

**Comment by Geolim4** [comment, marked as answer]
> **My question is , up-to when this data will be available in cached storage?**

Immediately. 
Default TTL is 900 second if left untouched.

**Reply by rahulradixweb** [reply]
Thanks for your response.

Means, If I get cache item (using getItem()) within 900 seconds, will it be extended (expiry time)?
And I don't get it within 900 seconds, it will be expired, is it so?
Any provision to override default TTL? without specifying to each write? (It seems , it can be done by setting global option defaultTtl when creating config, right?)

**Reply by rahulradixweb** [reply]
Do we have option to cache forever? untill we clear it explicitly?

**Reply by Geolim4** [reply]
> Means, If I get cache item (using getItem()) within 900 seconds, will it be extended (expiry time)? And I don't get it within 900 seconds, it will be expired, is it so? Any provision to override default TTL? without specifying to each write? (It seems , it can be done by setting global option defaultTtl when creating config, right?)

It's 900 seconds that start at the time you requested the item via getItem().
See global options in Wiki:
https://github.com/PHPSocialNetwork/phpfastcache/wiki/%5BV4%CB%96%5D-Configuration-Options#global-options

> Do we have option to cache forever? untill we clear it explicitly?

By definition a cache is not meant to be permanent, if you need permanent data use a database instead.
